### ISSUE TITLE
Teacher (resnet26) best accuracy

### ISSUE THREAD

**Issue by TeerathChandani** [imirzadeh/Teacher-Assistant-Knowledge-Distillation#8]
Hello Imirzadeh, 

Can you tell me the best accuracy of teacher? 

If you can share weight file of teacher, it would be better? 

Thanks.

**Comment by imirzadeh** [comment]
The validation accuracy of the teacher is 92.48

Here is the weight file:
[https://www.dropbox.com/s/bwsj585824hwd39/resnet26_d9cc8dae25984e2b9829d080c9c0e024_best.pth.tar?dl=0](https://www.dropbox.com/s/bwsj585824hwd39/resnet26_d9cc8dae25984e2b9829d080c9c0e024_best.pth.tar?dl=0)

**Comment by TeerathChandani** [comment]
But I am getting teacher accuracy that is almost 82%.. Why I am getting this low accuracy?

**Comment by imirzadeh** [comment]
What do you mean?
You are using my code to train a resnet 26 and getting 82% ?
If yes, what is your setting? how many epochs, ....?

**Comment by TeerathChandani** [comment]
Yes
Epoch : 160 
Seed(2) 

As other lambda and T are not used. So I just commented them. 

Thanks.

**Comment by TeerathChandani** [comment]
Secondly there was issue in 176 of train.py 

I replaced 

num_classes = 100 if dataset == 'cifar100' else 'cifar10'

 with 

num_classes = 100 if dataset == 'cifar100' else 10

**Comment by imirzadeh** [comment]
Hmm
Have you tried with another seed? Maybe it's a bad local optima.
If it still has the problem, let me know so I can debug it (hopefully this weekend)

**Comment by TeerathChandani** [comment]
Yes. I tried like 5, 20, and 50.  If we comment this seed, what do you think? 
I really do not know. why this is happening.. 

Thanks for your super fast reply. :) 

No issue take your time.  I am waiting till this weekends

**Comment by TeerathChandani** [comment]
By the way, which seed value your used for this training(teacher-resnet26)?

**Comment by imirzadeh** [comment]
Here is the complete execution for resnet26 training:
[https://www.comet.ml/cifar10/c10res26/d9cc8dae25984e2b9829d080c9c0e024](https://www.comet.ml/cifar10/c10res26/d9cc8dae25984e2b9829d080c9c0e024)

You can see the training charts in the "Charts" section on the left menu. The code is the dirty version of this repository since I cleaned the code before I put it on Github.

**Comment by TeerathChandani** [comment]
Thank you so much for your help. Let me try this.

**Comment by TeerathChandani** [comment]
Hello Dear Iman Mirzadeh,

Thanks for your help. I really got it. Did you train from Teacher(resnet26)
to student(resnet8) directly? I need that model, can you help me? I
trained, but I am getting very less accuracy almost 80% because may be I am
setting random parameters. You have nni lib and you can easily do it.

Again Thanks for providing assistance.

Waiting for your response.

Thanks,
Teerath Kumar

**Comment by TeerathChandani** [comment]
Hello Dear Iman Mirzadeh,

Thanks for your help. I really got it. Did you train from Teacher(resnet26)
to student(resnet8) directly? I need that model, can you help me? I
trained, but I am getting very less accuracy almost 80% because may be I am
setting random parameters. You have nni lib and you can easily do it.

Again Thanks for providing assistance.

And also can you provide weight file of below network for *CIFAR100*?
1. Res110
2. Res32
3. Res20

It would be really great if you provide.

Thanks,
Teerath Kumar

**Comment by imirzadeh** [comment]
Hello,

No. I trained using a teacher assistant(in two steps)
But, unfortunately, I'm extremely busy until September 5th and after that, I can share the details with you. I'm really sorry but I have too many works to do in the coming weeks.

Thanks,
Iman

**Comment by TeerathChandani** [comment]
Thanks, I will wait till 5th September
